src/DataProtection/DataProtection/src/AuthenticatedEncryption/AuthenticatedEncryptorFactory.cs

```diff
@@ -54,6 +55,9 @@ public AuthenticatedEncryptorFactory(ILoggerFactory loggerFactory)
 
             if (IsGcmAlgorithm(authenticatedConfiguration.EncryptionAlgorithm))
             {
+#if NETCOREAPP
+                return new AesGcmAuthenticatedEncryptor(new Secret(secret), GetAlgorithmKeySizeInBits(authenticatedConfiguration.EncryptionAlgorithm) / 8);
```

Comment:
Nit: I'd recommend not duplicating the secret here, and instead doing it in the class ctor. That way, regardless of how the encryptor object is created, it always owns its own secret.